src/main/resources/com/puppycrawl/tools/checkstyle/api/messages.properties

```diff
@@ -0,0 +1 @@
+import.ordering=Wrong order for ''{0}'' import.
```

Comment:
1. Why do we include message properties specific to `import.ordering` only into main resources but not the test ones?
2. Do we need all the locals to do testing or it's just enough to use a single message properties to test the case?

Reply:
1) I have just followed all the check, to throw the message in all the check it is on the resources of main, so I also put it in the resource section of main

2) `import.ordering` is only one which is require to throw the violation message at least for the test which I write so single message property is fine

Reply:
1. But it's a copy paste of existing properties into a different directory under main (non test) resources. @romani why do we allow this?
2. Why do you need all those `messages_XX.properties`, shouldn't a single file be enough?

Reply:
@rdiachenko for second question 
```
[INFO] 
[INFO] --- maven-antrun-plugin:3.1.0:run (ant-phase-verify) @ checkstyle ---
[INFO] Executing tasks
[INFO]      [echo] Checkstyle started (checkstyle-checks.xml): 22/06/2023 06:47:12 pm
[INFO] [checkstyle] Running Checkstyle  on 1626 files
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_de.properties' is missing. [Translation]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_fi.properties' is missing. [Translation]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_ru.properties' is missing. [Translation]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_pt.properties' is missing. [Translation]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_ja.properties' is missing. [Translation]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_fr.properties' is missing. [Translation]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_es.properties' is missing. [Translation]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_tr.properties' is missing. [Translation]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/api:1: Properties file 'messages_zh.properties' is missing. [Translation]
[INFO]      [echo] Checkstyle finished (checkstyle-checks.xml) : 22/06/2023 06:48:19 pm

```

it give error

Reply:
@romani ping

Reply:
No changes should done in src/main/resources folder at all.

Reply:
Considering that `FullIdent` is used by other checks, it should already be covered by the tests of those checks. I just did what pitest suggested, removed `.getFirstChild()` and run tests in `EqualsHashCodeCheckTest`. Many of them failed.

@romani why can't we make that test as part of pitest-api profile?

@Kevin222004 could you try to extend `pitest-api` profile with additional test class in the pom.xml?
```
git diff pom.xml 
diff --git a/pom.xml b/pom.xml
index 6bbbdfc62..92b07ca5d 100644
--- a/pom.xml
+++ b/pom.xml
@@ -4604,6 +4604,7 @@
                 <param>com.puppycrawl.tools.checkstyle.api.*</param>
                 <!-- 1% mutation in FullIdent -->
                 <param>com.puppycrawl.tools.checkstyle.checks.coding.UnusedLocalVariableCheckTest</param>
+                <param>com.puppycrawl.tools.checkstyle.checks.coding.EqualsHashCodeCheckTest</param>
                 <!-- 1% mutation in CheckUtil -->
                 <param>com.puppycrawl.tools.checkstyle.checks.OuterTypeFilenameCheckTest</param>
                 <param>org.checkstyle.suppressionxpathfilter.XpathRegressionUnusedLocalVariableTest</param>
```

Reply:
Yes sure I will do that

Reply:
I also not have any issue to following thing that i done. 

if everyone will not agree on https://github.com/checkstyle/checkstyle/pull/13179#discussion_r1248873846

> No changes should done in src/main/resources folder at all.

I update this soon

Reply:
This class is not API. Unfortunately leaked a long time ago. It is pure implementation and code in this implementation is not required by any of our Checks , it should be removed.

One day in future we will remove this class from API, it should be just another internal class.

Reply:
Done